java/c/src/test/java/org/apache/arrow/c/RoundtripTest.java

```diff
@@ -951,6 +952,130 @@ public void testImportReleasedArray() {
     }
   }
 
+  private FieldVector getSlicedVector(FieldVector vector, int offset) {
+    // Consumer allocates empty structures
+    try (ArrowSchema consumerArrowSchema = ArrowSchema.allocateNew(allocator);
+        ArrowArray consumerArrowArray = ArrowArray.allocateNew(allocator)) {
+
+      // Producer creates structures from existing memory pointers
+      try (ArrowSchema arrowSchema = ArrowSchema.wrap(consumerArrowSchema.memoryAddress());
+          ArrowArray arrowArray = ArrowArray.wrap(consumerArrowArray.memoryAddress())) {
+        // Producer exports vector into the C Data Interface structures
+        Data.exportVector(allocator, vector, null, arrowArray, arrowSchema);
+      }
+      // consumerArrowArray.snapshot().offset = offset;
```

Comment:
this is commented out?

Reply:
I am still working on updating the test case as you recommended. Meaning, as you have mentioned again here `manually tweaking snapshot` is what we have to do. I am still working on it. I just pushed so that I can work from my other device.

Reply:
Sorry about the confusion. Still addressing reviews.